### PR TITLE
Add a shard validator extension

### PR DESCRIPTION
**Pull request by sgkim126**

### CODE REVIEW

**Comment by sgkim126** [comment]
@kseo Would you review 343dbad18c13841397126b5b4de78bfefd0be369? The commit adds command line options for shard validator.

**Comment by sgkim126** [comment]
@joojis Would you review 7fec7d815a5d52f23fcacb19139854cd85fdb713? The commit adds a new RPC client, ShardValidatorClient.

**Comment by sgkim126** [comment]
@joojis And I have a question. I made register_action receives `Bytes` instead of `Action` since `Action` depends on many types that are not deserializable.
https://github.com/CodeChain-io/codechain/pull/488/commits/7fec7d815a5d52f23fcacb19139854cd85fdb713#diff-26b15409a957190592e4d3818ec8efa7R27
Is there some easy way to make `Action` derive `Deserialize`?

**Comment by joojis** [comment]
@sgkim126 One possibility is just to add the `Deserialize` macro to all the types that Action depends on.

**Comment by sgkim126** [comment]
@kseo I make it check `config.shard_validator.disable` before call `new_shard_validator`.

**Comment by sgkim126** [comment]
@joojis I make all types that Action depends on derive `Deserialize`

File: codechain/config/mod.rs [hunk 1]
```diff
@@ -325,6 +345,23 @@ impl Stratum {
...
 }
 
+impl ShardValidator {
+    pub fn overwrite_with(&mut self, matches: &clap::ArgMatches) -> Result<(), String> {
+        if matches.is_present("no-shard-validator") {
+            self.disable = true;
+        }
+
+        if let Some(account) = matches.value_of("shard-validator") {
+            self.account = Some(Address::from_str(account).map_err(|_| "Invalid address")?);
```
Comment (joojis) [inline codechain/config/mod.rs]: Why don't you use `parse_address()` here? It supports 0x-prefixed addresses.

Comment (sgkim126) [reply codechain/config/mod.rs]: I'll update it.

**Comment by sgkim126** [comment]
@joojis I added the RPC spec also.

**Comment by sgkim126** [comment]
@kseo As you said, I removed Encodable/Decodable for Signature.

**Pull request merged**